### PR TITLE
Is neighbor updated

### PR DESCRIPTION
**Pull request by anyaevostinar**
Adds support for getting all living organism neighbors in Mixed and Grid worlds including testing.

Gr, failing something other than documentation suddenly now...

### CODE REVIEW

File: include/emp/Evolve/World.hpp [hunk 1]
```diff
@@ -831,6 +839,9 @@ namespace emp {
     /// Use the specified function to get a neighbor (if not set, assume well mixed).
     WorldPosition GetRandomNeighborPos(WorldPosition pos) { return fun_get_neighbor(pos); }
 
+    /// Use the specified function to determine if two indices are neighboring.
+    bool IsNeighbor(size_t id1, size_t id2) {return fun_is_neighbor(id1, id2); }
```
Comment (amlalejini) [inline include/emp/Evolve/World.hpp:843]: unimportant style comment: inconsistent spacing

Comment (amlalejini) [inline include/emp/Evolve/World.hpp:843]: As above, should `size_t` arguments be of type `WorldPosition` here? @mercere99

File: include/emp/Evolve/World.hpp [hunk 2]
```diff
@@ -850,6 +861,8 @@ namespace emp {
     /// Return IDs of all empty cells in the population.
     emp::vector<size_t> GetEmptyPopIDs() { return FindCellIDs([](ORG*org){ return !org; }); }
 
+    /// Return IDs of all occupied neighbors of specified position.
+    emp::vector<size_t> GetValidNeighborOrgIDs(size_t id);
```
Comment (amlalejini) [inline include/emp/Evolve/World.hpp:865]: `size_t` vs `WorldPosition`?

File: include/emp/Evolve/World.hpp [hunk 3]
```diff
@@ -1342,6 +1386,20 @@ namespace emp {
...
 
+  /// Return IDs of all occupied neighbors of specified position.
+  template<typename ORG>
+  emp::vector<size_t> World<ORG>::GetValidNeighborOrgIDs(size_t id) { 
+    //Note: this function is similar to FindCellIDs, but because ORG don't know their index, 
+    //FindCellIDs can't be used.
+    emp::vector<size_t> valid_IDs(0);
+    for(size_t i = 0; i < pop.size(); i++) {
+      if ((bool) (pop[i].Raw()) && IsNeighbor(id, i)) {
+	valid_IDs.push_back(i);
```
Comment (amlalejini) [inline include/emp/Evolve/World.hpp:1397]: unimportant style comment: if statement contents not tabbed over

File: include/emp/Evolve/World.hpp [hunk 4]
```diff
@@ -1342,6 +1386,20 @@ namespace emp {
...
   }
 
+  /// Return IDs of all occupied neighbors of specified position.
+  template<typename ORG>
+  emp::vector<size_t> World<ORG>::GetValidNeighborOrgIDs(size_t id) { 
+    //Note: this function is similar to FindCellIDs, but because ORG don't know their index, 
+    //FindCellIDs can't be used.
+    emp::vector<size_t> valid_IDs(0);
+    for(size_t i = 0; i < pop.size(); i++) {
+      if ((bool) (pop[i].Raw()) && IsNeighbor(id, i)) {
```
Comment (amlalejini) [inline include/emp/Evolve/World.hpp:1396]: Would checking `IsOccupied` be equivalent to checking `pop[i].Raw()`?

**Review by amlalejini — CHANGES_REQUESTED** [review]
This PR adds useful functionality to the World class: IsNeighbor, GetValidNeighborOrgIDs, and SetIsNeighborFun. 

The travis build failures are not related to this PR (as discussed on slack). 

Overall, looks good to me. This could get merged now without issue. 

My comments mostly fall into two categories:
1.  minor style comments that don't super matter
2. There are a few places where `size_t` is used for an argument where `WorldPosition` might be more appropriate to be consistent with other World functions? @mercere99 should be able to chime in on this.

File: include/emp/Evolve/World.hpp [hunk 5]
```diff
@@ -131,6 +131,9 @@ namespace emp {
     /// Function type for identifying an organism's random neighbor.
     using fun_get_neighbor_t    = std::function<WorldPosition(WorldPosition)>;
 
+    /// Function type for determining if two organisms are neighbors.
+    using fun_is_neighbor_t     = std::function<bool(size_t, size_t)>;
```
[the remark this answers is not in the document]
Comment (anyaevostinar) [reply include/emp/Evolve/World.hpp:135]: When I tried to do it with WorldPosition I ran into an issue but now of course I can't remember what that issue was.

File: include/emp/Evolve/World.hpp (hunk 4 above)
Comment (anyaevostinar) [reply include/emp/Evolve/World.hpp:1396]: Quite possibly, this is modeled off of FindCellIDs

**Comment by anyaevostinar** [comment]
Changes made

**Comment by anyaevostinar** [comment]
Changes requested have been made!

**Review by amlalejini — APPROVED** [review]

**Comment by amlalejini** [comment]
`size_t` arguments swapped out for `WorldPosition` arguments to be consistent with the rest of the World class. 

Looks good to me!

**Comment by emilydolson** [comment]
This looks really useful! Is there a reason it isn't merged yet?

**Comment by anyaevostinar** [comment]
Just needed another approval, thank you!

**Pull request merged**